src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,454 @@
+  /**
+   * Optional label for the minimum value. If undefined then the min
+   * value is shown. Use empty string for no label.
+   */
+  minLabel?: React.ReactNode;
+  /**
+   * Optional label for the maximum value. If undefined then the max
+   *  value is shown. Use empty string for no label.
+   */
+  maxLabel?: React.ReactNode;
+  /**
+   * Defines the allowed behavior when moving Thumbs when multiple Thumbs are
+   * shown. It controls if a Thumb movement should be limited to only move in
+   * the segments adjacent to the Thumb. Possible values:
+   * 'allow-crossing' - allows thumb to cross other thumbs. Default.
+   * 'inhibit-crossing'- keeps the thumb from crossing and separated by a step.
+   * @default 'inhibit-crossing'
+   */
+  thumbMode?: 'allow-crossing' | 'inhibit-crossing';
+  /**
+   * Additional props to include on each thumb container.
+   */
+  thumbProps?: Omit<CommonProps, 'title'>;
+  /**
+   * Callback fired when the value(s) of the slider has changed. This will receive
+   * changes at the end of a slide as well as changes from clicks on rails and tracks.
+   */
+  onChange?: (values: ReadonlyArray<number>) => void;
+  /**
+   * Callback fired when the value(s) of the slider are internally updated during
+   * operations like dragging a Thumb. Use this callback with caution as a
+   * high-volume of updates will occur when dragging.
+   */
+  onUpdate?: (values: ReadonlyArray<number>) => void;
+} & Omit<CommonProps, 'title'>;
+
+/**
+ * Slider component that display Thumbs for each value specified along a Rail.
+ * @example
+ * <Slider values={[10]} min={0} max={60} disabled />
+ * <Slider values={[10, 20]} min={0} max={50} step={2} />
+ * <Slider values={[10, 20, 30, 40]} min={0} max={60} setFocus
+ *   thumbMode='allow-crossing' />
+ */
+export const Slider = React.forwardRef<HTMLDivElement, SliderProps>(
+  (props, ref) => {
+    const {
+      min = 0,
+      max = 100,
+      values,
+      step = 1,
+      setFocus = false,
+      tooltipProps,
+      tooltipRenderer,
+      disabled = false,
+      tickLabels,
+      minLabel,
+      maxLabel,
+      trackDisplayMode = undefined,
+      thumbMode = 'inhibit-crossing',
+      onChange,
+      onUpdate,
+      thumbProps,
+      className,
+      ...rest
+    } = props;
+
+    const [currentValues, setCurrentValues] = React.useState(values);
+    React.useEffect(() => {
+      setCurrentValues(values);
+    }, [values]);
+
+    const [minValueLabel, setMinValueLabel] = React.useState(minLabel);
+    React.useEffect(() => {
+      setMinValueLabel(minLabel ?? min.toString());
+    }, [minLabel, min]);
+
+    const [maxValueLabel, setMaxValueLabel] = React.useState(maxLabel);
+    React.useEffect(() => {
+      setMaxValueLabel(maxLabel ?? max.toString());
+    }, [maxLabel, max]);
+
+    const [trackDisplay, setTrackDisplay] = React.useState<TrackDisplayMode>(
+      () => getDefaultTrackDisplay(trackDisplayMode, currentValues),
```

Comment:
Arrow fun needed? Cant just call `getDefaultTrackDisplay(trackDisplayMode, currentValues)` ?

Reply:
From what I was told. in some Pluralsight course, this method is supposed to keep the getDefaultTrackDisplay function from being called on every render and would only get called when mounted.

Reply:
Yup, here's the docs: https://reactjs.org/docs/hooks-reference.html#lazy-initial-state

Reply:
ok, didnt know. Nice.